plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -192,11 +176,11 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     @Override
     public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaNameOrNull)
     {
-        ImmutableList.Builder<SchemaTableName> builder = ImmutableList.builder();
-        for (String table : getPinotTableNames()) {
+        ImmutableSet.Builder<SchemaTableName> builder = ImmutableSet.builder();
+        for (String table : pinotClient.getPinotTableNames()) {
             builder.add(new SchemaTableName(SCHEMA_NAME, table));
         }
-        return builder.build();
+        return ImmutableList.copyOf(builder.build());
```

Comment:
I expected Modernizer to suggest using `buildOrThrow` instead of `build`. Note to self to check.

Reply:
Ah thanks, it doesn't have any reason to. It's just for ImmutableMap since it's not obvious it would throw on duplicate entries. For lists and sets the only thing is that they are null-hostile, duplicates are ignored in ImmutableSet.